java/vector/src/main/java/org/apache/arrow/vector/sort/VectorSorter.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.arrow.vector.sort;
+
+import org.apache.arrow.vector.ValueVector;
+
+/**
+ * Basic interface for sorting a vector.
+ * @param <V> the vector type.
+ */
+public interface VectorSorter<V extends ValueVector> {
+
+  /**
+   * Sort the vector by the given criteria.
+   * The sort can be in-place or out-of-place, depending on the algorithm.
```

Comment:
Is there a reason you chose this interface instead of separating it into two (InplaceVectorSorter and OutOfPlaceVectorSorter)?   Or at least two different methods?

Reply:
Good question.

I think it is important that there be a uniform way of sorting a vector, so that for some scenarios, the underlying sorting algorithms can be hidden from the caller of the sorter.

The uniform sorting API means:
1. There must be a common base class/interface for the sorters.
2. There should be a uniform sorting method.

In the new commits, I have added a method to the interface (isInPlace) to distinguish between in-place and out-of-place sorters.